pkg/policy/rules.go

```diff
@@ -0,0 +1,158 @@
+package policy
+
+import (
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/filters"
+	"github.com/aquasecurity/tracee/pkg/utils"
+)
+
+type RulesData struct {
+	Data                        map[events.ID]RuleData
+	kernelDataFilterMatchStates KernelDataFilterMatchStates
```

Comment:
I don't think we should add this extra Data hierarchy level.
This is related to the fact that you added `kernelDataFilterMatchStates` in here, but I think this is not the correct abstraction level to add it into.
The current policy struct represents the way a policy in Tracee looks like:

```
type Policy struct {
        **// Policy metadata:**
	ID                int
	Name              string
        **// All scope filters are below:**
	UIDFilter         *filters.UIntFilter[uint32]
	PIDFilter         *filters.UIntFilter[uint32]
	NewPidFilter      *filters.BoolFilter
	MntNSFilter       *filters.UIntFilter[uint64]
	PidNSFilter       *filters.UIntFilter[uint64]
	UTSFilter         *filters.StringFilter
	CommFilter        *filters.StringFilter
	ContFilter        *filters.BoolFilter
	NewContFilter     *filters.BoolFilter
	ContIDFilter      *filters.StringFilter
	ProcessTreeFilter *filters.ProcessTreeFilter
	BinaryFilter      *filters.BinaryFilter
	Follow            bool
        **// All rules are grouped, just like in a policy of Tracee**
	Rules             map[events.ID]RuleData
}

type RuleData struct {
	EventID     events.ID
        **// All the per-rule filters are here:**
	ScopeFilter *filters.ScopeFilter
	DataFilter  *filters.DataFilter
	RetFilter   *filters.IntFilter[int64]
}
```

So, I think the kernel data filter logic should be pushed to a lower level object, probably during kernel filter map calculation in ebpf.go

Reply:
Good ideia. I did another version where I called a new function`computeDataFilterEqualities` inside the function `computeFilterEqualities` to compute kernel data filter. The rest of the logic remains the same.

https://github.com/aquasecurity/tracee/blob/f8b46e634dade61fdac21d0d65d4c580016f5099/pkg/policy/equality.go#L227

After 1) computing kernel data filter equalities, we proceed to 2) generate the config_map. To do this, we need to determine if at least one of the following types was set: exact, not exact, prefix, not prefix, suffix, or not suffix. Since we already iterate over all policies in step 1, I added a field to the policies struct to store this computed data. WDYT?